cowswap_python_client/cow_client.py

```diff
@@ -0,0 +1,123 @@
+from typing import Any
+
+import requests
+from eth_account import Account
+from eth_account.signers.local import LocalAccount
+from eth_typing import ChecksumAddress
+from loguru import logger
+from web3 import Web3
+
+from cowswap_python_client.encoding import (
+    DOMAIN,
+    MESSAGE_TYPES,
+    MESSAGE_TYPES_CANCELLATION,
+)
+from cowswap_python_client.gtypes import Wei
+from cowswap_python_client.models import (
+    CowServer,
+    OrderKind,
+    OrderStatus,
+    QuoteInput,
+    QuoteOutput,
+)
+
+
+class CowClient:
+    def __init__(
+        self, account: LocalAccount, api_url: CowServer = CowServer.GNOSIS_STAGING
+    ):
+        self.api_url = api_url
+        self.account = account
+
+    def get_version(self) -> str:
+        r = requests.get(f"{self.api_url.value}/api/v1/version")
+        return r.text
+
+    def build_swap_params(
+        self, sell_token: ChecksumAddress, buy_token: ChecksumAddress, sell_amount: Wei
+    ) -> QuoteInput:
+        quote = QuoteInput(
+            from_=self.account.address,
+            sell_token=sell_token,
+            buy_token=buy_token,
+            receiver=self.account.address,
+            sell_amount_before_fee=str(sell_amount),
+            kind=OrderKind.SELL,
+            app_data="0x0000000000000000000000000000000000000000000000000000000000000000",
+            valid_for=1080,
+        )
+        return quote
+
+    @staticmethod
+    def _if_error_log_and_raise(r: requests.Response) -> None:
+        try:
+            r.raise_for_status()
+        except:
+            logger.error(f"Error occured on response: {r.text}")
+            r.raise_for_status()
```

Comment:
```suggestion
            raise
```

No need to commitl, just fyi that you can just do `raise` to re-raise an exception in except block.

Reply:
Thanks, added this - it's nice because the exception traceback is kept.